noir-projects/noir-protocol-circuits/crates/rollup-lib/src/components.nr

```diff
@@ -166,10 +166,7 @@ pub fn compute_tx_effects_hash(
     let new_l2_to_l1_msgs = combined.new_l2_to_l1_msgs;
 
     // Public writes are the concatenation of all non-empty user update requests and protocol update requests, then padded with zeroes.
-    let public_data_update_requests = get_all_update_requests_for_tx_effects(
-        combined.public_data_update_requests,
-        protocol_public_data_update_requests
-    );
+    let public_data_update_requests = get_all_update_requests_for_tx_effects(all_public_data_update_requests);
```

Comment:
Do we still need to call this function to transform the array? Think it should already be padded with empty items.

Reply:
Sorry I didn't respond to this earlier! But it looks like `get_all_update_requests_for_tx_effects` takes an array of size N and outputs an array of size N. If the input array already has all the non-empty items at the beginning and padded with empty items, which I believe is the case, the output array should be exactly the same as the input 🤔

Reply:
Ahh good point! Still, there is a difference: this function ensures that all non-empty requests are pushed to the end of the array, which is needed for consistency with the solidity implementation.

Reply:
I'll add a comment on a next PR to make it clear!